extensions/amp-sidebar/0.1/amp-sidebar.js

```diff
@@ -103,6 +107,19 @@ export class AmpSidebar extends AMP.BaseElement {
     this.registerAction('toggle', this.toggle_.bind(this));
     this.registerAction('open', this.open_.bind(this));
     this.registerAction('close', this.close_.bind(this));
+
+    installGlobalClickListenerForDoc(this.getAmpDoc()).addBeforeHandler(e => {
```

Comment:
> Why do we need to listen for global clicks anyway?

This is basically to make sure we run before we actually handle the click (which is handled inside `document-click` handler), and the event is captured there and from testing capturing seems to be first-capture-first-serve and since `document-click` listener is added on document load it's always going to be the first handle to be called. 

In this case we want our handler inside sidebar to be called before it to allow us to pop before pushing.